microprofile-config/src/main/java/org/jboss/eap/qe/microprofile/config/testapp/jaxrs/ResolverEndPoint.java

```diff
@@ -0,0 +1,108 @@
+    public String configSourceInOneConfig() {
+        // prepare response
+        StringBuilder response = new StringBuilder();
+
+        // get resolver
+        ConfigProviderResolver resolver = ConfigProviderResolver.instance();
+
+        // handle original config
+        Config originalConfig = resolver.getConfig();
+        response.append(originalConfig.getValue(PROPERTY_NAME, String.class));
+        resolver.releaseConfig(originalConfig);
+
+        // prepare, register and release new config
+        ConfigBuilder builder = resolver.getBuilder();
+        Config specificConfig = builder.addDefaultSources().withSources(new ResolverConfigSource()).build();
+        resolver.registerConfig(specificConfig, ResolverEndPoint.class.getClassLoader());
+        Config customConfig = resolver.getConfig();
+        response.append(customConfig.getValue(PROPERTY_NAME, String.class));
+        resolver.releaseConfig(customConfig);
+
+        // get default config
+        originalConfig = resolver.getConfig();
+        response.append(originalConfig.getValue(PROPERTY_NAME, String.class));
+
+        return response.toString();
+    }
+
+    /**
+     * This end point checks behaviour of two config
+     *
+     * This behaviour may be changed in future spec release. See https://github.com/eclipse/microprofile-config/issues/522
+     */
+    @GET
+    @Path("/twoConfigSources")
+    public String configSourceInTwoConfig() {
+
+        // prepare response
+        StringBuilder response = new StringBuilder();
+
+        // get resolver
+        ConfigProviderResolver resolver = ConfigProviderResolver.instance();
+
+        // handle original config
+        Config originalConfig = resolver.getConfig();
+        resolver.releaseConfig(originalConfig);
+
+        // prepare two configs
+        ConfigBuilder builder = resolver.getBuilder();
+        ConfigSource customConfigSource = new ResolverConfigSource();
+        Config specificConfig1 = builder.withSources(customConfigSource).build();
+        Config specificConfig2 = builder.withSources(customConfigSource).build();
+
+        // register first config
+        resolver.registerConfig(specificConfig1, ResolverEndPoint.class.getClassLoader());
+
+        // try to register second config, exception is expected
+        try {
+            resolver.registerConfig(specificConfig2, ResolverEndPoint.class.getClassLoader());
+            return ILLEGAL_STATE_EXCEPTION_ERROR_MESSAGE;
+        } catch (IllegalStateException ise) {
+            if (!ise.getMessage().equals("Configuration already registered for the given class loader")) {
+                return ILLEGAL_STATE_EXCEPTION_ERROR_MESSAGE;
+            }
+        }
+
+        // get value from first config
+        Config customConfig1 = resolver.getConfig();
+        response.append(customConfig1.getValue(PROPERTY_NAME, String.class));
+        resolver.releaseConfig(customConfig1);
+
+        // get value from second config
```

Comment:
@jmesnil Based on our f2f mtg, I'm adding second test that checks behaviour of two Config objects with the same ConfigSource object. This test checks that second Config object can't be registered by registerConfig. This test also checks property values from both Config objects.

Reply:
@jmesnil Are you ok with this PR now?